pkg/identityManager/iamIdentityProvider.go

```diff
@@ -0,0 +1,189 @@
+package identitymanager
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"gopkg.in/yaml.v3"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	responsetypes "github.com/liqotech/liqo/pkg/identityManager/responseTypes"
+)
+
+const (
+	namespaceKubeSystem      = "kube-system"
+	awsAuthConfigMapName     = "aws-auth"
+	awsAuthConfigMapUsersKey = "mapUsers"
+)
+
+type iamIdentityProvider struct {
+	awsConfig *AwsConfig
+	client    kubernetes.Interface
+}
+
+type mapUser struct {
+	UserArn  string   `json:"userarn"`
+	Username string   `json:"username"`
+	Groups   []string `json:"groups"`
+}
+
+func (identityProvider *iamIdentityProvider) GetRemoteCertificate(clusterID,
+	signingRequest string) (response responsetypes.SigningRequestResponse, err error) {
+	// this method has no meaning for this identity provider
+	return response, kerrors.NewNotFound(schema.GroupResource{
+		Group:    "v1",
+		Resource: "secrets",
+	}, remoteCertificateSecret)
+}
+
+func (identityProvider *iamIdentityProvider) ApproveSigningRequest(clusterID,
+	signingRequest string) (response responsetypes.SigningRequestResponse, err error) {
+	sess := session.Must(session.NewSession(&aws.Config{
```

Comment:
What does `Must` imply. Does it panic in case of failure?

Reply:
Yes, exactly. When I wrote that I was sure that was a good idea to not hide authentication problems with amazon and to have a clear restart of the component, but now I'm changing my idea 😅

Reply:
ok, I removed that